lib/Db/ItemMapperV2.php

```diff
@@ -56,9 +56,10 @@ public function findAllFromUser(string $userId, array $params = []): array
         $builder->select('items.*')
                 ->from($this->tableName, 'items')
                 ->innerJoin('items', FeedMapperV2::TABLE_NAME, 'feeds', 'items.feed_id = feeds.id')
-                ->where('feeds.user_id = :user_id')
+                ->where('(feeds.user_id = :user_id AND items.shared_by = \'\') OR items.shared_with = :shared_with')
```

Comment:
I don't think shared articles should be in your normal list since we can't explain how they got there.

Reply:
I agree with @SMillerDev on that one.

Apart from that, the query could probably be simplified
```sql
feeds.user_id = :user_id OR items.shared_with = :user_id
```
The `:user_id` parameter ca be reused as it contains the same value anyways .. and from what I've seen so far `shared_by` is always empty if the feed's user matches.

Reply:
We've thought about this and for us it makes sense to include shared articles in this function as well as in findAllItems, since they should be shown to the user.

Also, these functions are used in other functions which need access to shared items. For example, if we star an item that is shared with us, we want it to appear in the "Starred" menu. To retrieve the starred items we use _findAllItems()_, and the _starred()_ function used for the starred count uses _findAllFromUser()_

The simplification that you've suggested wouldn't work in our implementation.
```sql
feeds.user_id = :user_id OR items.shared_with = :user_id
```
Here's why: if user A shares an item with user B, the item is still in user A's feed, with _shared_by_ = B.
If we want to find all items for user A, we need to verify that _shared_by_ is empty, otherwise it's an article that's shared with someone. That article should instead be shown to user B.